pkg/component/generic/http/v0/component_test.go

```diff
@@ -22,10 +27,211 @@ const (
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+
+		bodyBytes, _ := io.ReadAll(r.Body)
+
+		response := map[string]any{
+			"data":    string(bodyBytes),
+			"headers": r.Header,
+		}
+		_ = json.NewEncoder(w).Encode(response)
+	})
+
+	// DELETE /delete - Simple response
+	mux.HandleFunc("/delete", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodDelete {
+			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		response := map[string]any{
+			"url": r.URL.String(),
+		}
+		_ = json.NewEncoder(w).Encode(response)
+	})
+
+	// PUT /put - Echo request data
+	mux.HandleFunc("/put", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPut {
+			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+
+		var requestBody any
+		if strings.Contains(r.Header.Get("Content-Type"), "application/json") {
+			bodyBytes, _ := io.ReadAll(r.Body)
+			_ = json.Unmarshal(bodyBytes, &requestBody)
+		}
+
+		response := map[string]any{
+			"json":    requestBody,
+			"headers": r.Header,
+		}
+		_ = json.NewEncoder(w).Encode(response)
+	})
+
+	// GET /status/500 - Return 500 error
+	mux.HandleFunc("/status/500", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+			return
+		}
+		w.Header().Set("Content-Type", "text/html; charset=utf-8")
+		w.WriteHeader(500)
+	})
+
+	// Basic auth endpoints
+	mux.HandleFunc("/basic-auth/foo/bar", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+			return
+		}
+
+		username, password, ok := r.BasicAuth()
+		if !ok || username != "foo" || password != "bar" {
+			w.WriteHeader(401)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		response := map[string]any{
+			"authenticated": true,
+			"user":          username,
+		}
+		_ = json.NewEncoder(w).Encode(response)
+	})
+
+	// Bearer token endpoint
+	mux.HandleFunc("/bearer", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+			return
+		}
+
+		auth := r.Header.Get("Authorization")
+		if !strings.HasPrefix(auth, "Bearer ") {
+			w.WriteHeader(401)
+			return
+		}
+
+		token := strings.TrimPrefix(auth, "Bearer ")
+		w.Header().Set("Content-Type", "application/json")
+		response := map[string]any{
+			"authenticated": true,
+			"token":         token,
+		}
+		_ = json.NewEncoder(w).Encode(response)
+	})
+
+	return httptest.NewServer(mux)
+}
+
 func TestComponent(t *testing.T) {
 	c := qt.New(t)
 	c.Parallel()
 
+	// Set test environment to bypass URL validation
+	os.Setenv("GO_TESTING", "true")
```

Comment:
Nit: instead of relying on env variables I'd made the whitelisted URLs or the URL validation function a constructor param for the component, it makes the behaviour of the component more explicit.